modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/tree/BPlusTree.java

```diff
@@ -617,6 +604,28 @@ else if (needBackIfRouting) {
         }
     }
 
+    /**
+     * Page handler that adds the page to the tail of {@link Update} object.
+     * Results in {@link Result#FOUND} if added to tail successfully.
+     * Results in {@link Result#RETRY} if triangle invariant is violated.
```

Comment:
Is this "triangle invariant" some common knowledge? I could find any explanation in this class

Reply:
There's no explicit explanation, but it can be derived from most places when it is checked. If there's a good place for it - I'll add a comment

Reply:
I think the class' javadoc might be a suitable place